api/v2.0/swagger.yaml

```diff
@@ -2343,6 +2343,327 @@ paths:
           description: The API server is alive
           schema:
             type: string
+  /retentions/metadatas:
+    get:
+      summary: Get Retention Metadatas
+      description: Get Retention Metadatas.
+      operationId: getRentenitionMetadata
+      tags:
+        - Retention
+      responses:
+        '200':
+          description: Get Retention Metadatas successfully.
+          schema:
+            $ref: '#/definitions/RetentionMetadata'
+
+  /retentions:
+    post:
+      summary: Create Retention Policy
+      operationId: createRetention
+      description: >-
+        Create Retention Policy, you can reference metadatas API for the policy model.
+        You can check project metadatas to find whether a retention policy is already binded.
+        This method should only be called when no retention policy binded to project yet.
+      tags:
+        - Retention
+      parameters:
+        - name: policy
+          in: body
+          description: Create Retention Policy successfully.
+          required: true
+          schema:
+            $ref: '#/definitions/RetentionPolicy'
+      responses:
+        '201':
```

Comment:
Use the predefined responses instead: https://github.com/goharbor/harbor/blob/master/api/v2.0/swagger.yaml#L108